src/lib/fetchers/fetchPlacesOnKomootPhoton.ts

```diff
@@ -0,0 +1,72 @@
+import { Point } from 'geojson'
+import { t } from 'ttag'
+
+export type KomootPhotonResultProperties = {
+  city?: string;
+  country?: string;
+  name?: string;
+  osm_id?: number;
+  osm_key?: string;
+  osm_type?: 'N' | 'W' | 'R';
+  osm_value?: string;
+  postcode?: string;
+  countrycode?: string;
+  locality?: string;
+  type?: string;
+  district?: string;
+  state?: string;
+  housenumber?: string;
+  street?: string;
+  extent?: [number, number, number, number];
+};
+
+export type KomootPhotonResultFeature = {
+  geometry: Point;
+  properties: KomootPhotonResultProperties;
+};
+
+export type SearchResultCollection = {
+  features: KomootPhotonResultFeature[],
+  error?: any,
+};
+
+export default async function fetchPlacesOnKomootPhoton({ query, additionalQueryParameters = {} }: {
+  query: string,
+  additionalQueryParameters: Record<string, string | string[]> | {},
+}): Promise<SearchResultCollection> {
+  const languageCode = window.navigator.language.slice(0, 2)
+  const supportedLanguageCodes = ['en', 'de', 'fr', 'it'] // See Photon documentation
+
+  const queryParameters = new URLSearchParams([
+    ['q', query],
+    ['limit', '30'],
+    ...Object.entries(additionalQueryParameters)
+      .flatMap(
+        ([key, stringOrStringArray]) => (
+          typeof stringOrStringArray === 'string'
+            ? [[key, stringOrStringArray]]
+            : stringOrStringArray.map((value) => [key, value])
+        ),
+      ),
+  ])
+
+  if (supportedLanguageCodes.includes(languageCode)) {
+    queryParameters.append('lang', languageCode)
+  }
+
+  const url = `https://photon.komoot.io/api/?${queryParameters.toString()}`
```

Comment:
When entering "Berlin", does it find the most relevant Berlin? We could bias the result using the location from the country we get from the CDN (via IP).

Reply:
I'm not certain how to do it and can't find useful references how it's being used on other components.